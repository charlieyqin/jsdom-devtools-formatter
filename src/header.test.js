const header = require("./header");

describe("header", () => {
  it("returns null to utilize default formatter for a non-HTML elements", () => {
    expect(header()).toBe(null);
    expect(header(null)).toBe(null);
    expect(header(undefined)).toBe(null);
    expect(header(false)).toBe(null);
    expect(header("a string")).toBe(null);
    expect(header(123)).toBe(null);
    expect(header({})).toBe(null);
    expect(header([])).toBe(null);
    expect(header(new Date())).toBe(null);
  });

  it("returns summary for a simple DOM element", () => {
    const element = document.createElement("div");
    element.innerHTML = 'test text';
    expect(header(element)).toMatchSnapshot();
  });

  it("returns a ellipsis for a complicated DOM element", () => {
    const element = document.createElement("div");
    element.className = "foo bar baz";
    element.id = "root";
    element.innerHTML = `
      <h1>This is not part of the header</h1>
      <p class="example">And this neither</p>
    `;
    expect(header(element)).toMatchSnapshot();
  });
});
