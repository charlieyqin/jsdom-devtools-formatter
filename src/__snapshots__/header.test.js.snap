// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header returns a ellipsis for a complicated DOM element 1`] = `
Array [
  "span",
  Object {
    "style": "
    color: rgb(168, 148, 166);
  ",
  },
  "<",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(136, 18, 128);
  ",
    },
    "div",
  ],
  " ",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(153, 69, 0);
    unicode-bidi: -webkit-isolate;
  ",
    },
    "class",
  ],
  "=\\"",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(26, 26, 166);
    unicode-bidi: -webkit-isolate;
  ",
    },
    "foo bar baz",
  ],
  "\\"",
  " ",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(153, 69, 0);
    unicode-bidi: -webkit-isolate;
  ",
    },
    "id",
  ],
  "=\\"",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(26, 26, 166);
    unicode-bidi: -webkit-isolate;
  ",
    },
    "root",
  ],
  "\\"",
  ">",
  Array [
    "span",
    Object {
      "style": "
    color: black;
    unicode-bidi: -webkit-isolate;
  ",
    },
    "â€¦",
  ],
  "<",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(136, 18, 128);
  ",
    },
    "/div",
  ],
  ">",
]
`;

exports[`header returns summary for a simple DOM element 1`] = `
Array [
  "span",
  Object {
    "style": "
    color: rgb(168, 148, 166);
  ",
  },
  "<",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(136, 18, 128);
  ",
    },
    "div",
  ],
  ">",
  Array [
    "span",
    Object {
      "style": "
    color: black;
    unicode-bidi: -webkit-isolate;
  ",
    },
    "test text",
  ],
  "<",
  Array [
    "span",
    Object {
      "style": "
    color: rgb(136, 18, 128);
  ",
    },
    "/div",
  ],
  ">",
]
`;
